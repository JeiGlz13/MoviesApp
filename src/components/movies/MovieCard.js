import React from 'react';
import {
    Link
  } from "react-router-dom";

export const MovieCard = ({id, name, Director, Year, Cast}) => {
    return (
        <div className="p-0 mt-16 mx-5 items-center flex flex-wrap cardShadow rounded-3xl">
            <div className=" ml-auto mr-4">
                <img alt="..." className="rounded-lg shadow-lg h-80 " 
                    src = {`./assets/movies/${id}.jpg`}/>
            </div>

            <div className="w-72 py-0 pr-4">
                    <h3 className="text-xl font-semibold mb-1">{name}</h3>

                    <p className="text-md text-gray-500 font-semibold mb-4">
                        Release: {Year}
                    </p>
                    <ul className="list-none mt-2">
                        <li className="py-1">
                            <div className="flex items-center">
                                <div>
                                    <span className="text-xs font-semibold  py-2 px-2 uppercase rounded-full text-pink-600 bg-pink-200 mr-3">
                                        <i className="fa-solid fa-clapperboard text-md"></i>
                                    </span>
                                </div>
                                <div>
                                    <h4 className="text-gray-600 text-md">
                                        <strong>Director:</strong> {Director}
                                    </h4>
                                </div>
                            </div>
                        </li>
                        <li className="py-1">
                            <div className="flex items-center">
                                <div>
                                    <span className="text-xs font-semibold py-2 px-2 uppercase rounded-full text-pink-600 bg-pink-200 mr-3">
                                        <i className="fa-solid fa-user text-md"></i>
                                    </span>
                                </div>
                                <div>
                                    <h4 className="text-gray-600 text-md">
                                        <strong>Cast:</strong>  {Cast}
                                    </h4>
                                </div>
                            </div>
                        </li>
                    </ul>

                    <div className='mt-6' >
                        <Link to={`./movie/${id}`}
                        className = 'px-4 py-2 bg-red-500 text-white rounded-lg' >
                                Mas...
                        </Link>
                    </div>
            </div>
        </div>
    )
}
