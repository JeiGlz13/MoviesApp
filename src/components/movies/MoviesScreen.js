import React, { useMemo } from 'react'
import { Redirect, useParams } from 'react-router-dom'
import { getMovieById } from '../selectors/getMovieById';

export const MoviesScreen = ({history}) => {
    const {movieId} = useParams();

    const movie = useMemo(() => getMovieById(movieId), [movieId]);
     
    if(!movie){
        return <Redirect to="/"/>;
    }

    const {name, Genre, Director, Year, Cast} = movie;

  const handleReturn = ()=>{
      if(history.length <=2){
          history.push('/');
      }else{
        history.goBack();
      }
       
  }

    return (
        <div className ="">
            <div className = "">
                <img src={`../assets/movies/${movieId}.jpg`}
                className = " animate__animated animate__fadeIn"
                alt = {name} />
            </div>

            <div className = " animate__animated animate__fadeInUp">
                <h3>
                    {name}
                </h3>
                <ul className = "list-group list-group-flush">
                    <li className = "list-group-item">
                        <b>Director: {Director}</b>
                    </li>
                    <li className = "list-group-item">
                        <b>Genre: {Genre}</b>
                    </li>
                    <li className = "list-group-item">
                        <b>Release: {Year}</b>
                    </li>
                </ul>

                <h5>Cast</h5>
                <p>{Cast}</p>

                <button className = "btn btn-outline-info"
                onClick = {handleReturn}>
                    Return
                </button>
            </div>
        </div>
    )
}
