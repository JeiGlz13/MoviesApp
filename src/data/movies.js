export const movies = [
    {
        'id': 'gang-ABronxTale',
        'name':'A Bronx Tale', 
        'Genre':'Gangsters', 
        'Director':'Robert De Niro',
        'Year':'1993',
        'Cast':'Roberto De Niro, Chazz Palminteri, Lillo Brancato'
    },
    {
        'id': 'gang-Casino',
        'name':'Casino', 
        'Genre':'Gangsters', 
        'Director':'Martin Scorsese',
        'Year':'1995',
        'Cast':'Robert De Niro, Joe Pesci, Sharon Stone'
    },
    {
        'id': 'gang-Goodfellas',
        'name':'GoodFellas', 
        'Genre':'Gangsters', 
        'Director':'Martin Scorsese',
        'Year':'1990',
        'Cast':'Robert De Niro, Joe Pesci, Ray Liotta'
    },
    {
        'id': 'gang-OnceUponATimeInAmerica',
        'name':'Once Upon A Time In America', 
        'Genre':'Gangsters', 
        'Director':'Sergio Leone',
        'Year':'1984',
        'Cast':'Robert De Niro, James Woods, Elizabeth McGovern'
    },
    {
        'id': 'gang-Scarface',
        'name':'Scarface', 
        'Genre':'Gangsters', 
        'Director':'Brian De Palma',
        'Year':'1983',
        'Cast':'Al Pacino, Michelle Pfeiffer, Steven Bauer'
    },
    {
        'id': 'gang-TheGodfather',
        'name':'The Godfather: Part I', 
        'Genre':'Gangsters', 
        'Director':'Francis Ford Coppola',
        'Year':'1972',
        'Cast':'Marlon Brando, Al Pacino, James Caan'
    },
    {
        'id': 'gang-TheGodfatherPartII',
        'name':'The Godfather: Part II', 
        'Genre':'Gangsters', 
        'Director':'Francis Ford Coppola',
        'Year':'1974',
        'Cast':'Al Pacino, Robert De Niro, Robert Duvall'
    },
    {
        'id': 'gang-TheGodfatherPartIII',
        'name':'The Godfather: Part III', 
        'Genre':'Gangsters', 
        'Director':'Francis Ford Coppola',
        'Year':'1990',
        'Cast':'Al Pacino, Andy Garcia, Diane Keaton'
    },
    {
        'id': 'gang-TheIrishman',
        'name':'The Irishman', 
        'Genre':'Gangsters', 
        'Director':'Martin Scorsese',
        'Year':'2019',
        'Cast':'Robert De Niro, Al Pacino, Joe Pesci'
    },
    {
        'id': 'gang-TheUntouchables',
        'name':'The Untouchables', 
        'Genre':'Gangsters', 
        'Director':'Brian De Palma',
        'Year':'1987',
        'Cast':'Kevin Costner, Sean Connery, Robert De Niro'
    },
    {
        'id': 'war-1917',
        'name':'1917', 
        'Genre':'War', 
        'Director':'Sam Mendes',
        'Year':'2019',
        'Cast':' George MacKay, Dean-Charles Chapman'
    },
    {
        'id': 'war-ApocalypseNow',
        'name':'Apocalypse Now', 
        'Genre':'War', 
        'Director':'Francis Ford Coppola',
        'Year':'1979',
        'Cast':'Marlon Brando, Robert Duvall, Martin Sheen'
    },
    {
        'id': 'war-Dunkirk',
        'name':'Dunkirk', 
        'Genre':'War', 
        'Director':'Christopher Nolan',
        'Year':'2017',
        'Cast':'Fionn Whitehead, Tom Hardy, Cillian Murphy'
    },
    {
        'id': 'war-FullMetalJacket',
        'name':'Full Metal Jacket', 
        'Genre':'War', 
        'Director':'Stanley Kubrick',
        'Year':'1987',
        'Cast':"Matthew Modine, R. Lee Ermey, Vincent D'Onofrio"
    },
    {
        'id': 'war-GraveOfTheFireflies',
        'name':'Grave of The Fireflies', 
        'Genre':'War', 
        'Director':'Isao Takahata',
        'Year':'1988',
        'Cast':'Tsutomu Tatsumi, Ayano Shiraishi, Yoshiko Shinohara'
    },
    {
        'id': 'war-MyNameIsIvan',
        'name':'My Name is Ivan', 
        'Genre':'War', 
        'Director':'Andrei Tarkovsky',
        'Year':'1962',
        'Cast':'Nikolai Burlyayev, Valentin Zubkov '
    },
    {
        'id': 'war-PathsOfGlory',
        'name':'Paths of Glory', 
        'Genre':'War', 
        'Director':'Stanley Kubrick',
        'Year':'1957',
        'Cast':'Kirk Douglas, Adolphe Menjou'
    },
    {
        'id': 'war-SavingPrivateRyan',
        'name':'Saving Private Ryan', 
        'Genre':'War', 
        'Director':'Steven Spielberg',
        'Year':'1998',
        'Cast':'Tom Hanks, Matt Damon, Tom Sizemore'
    },
    {
        'id': 'war-SchindlersList',
        'name': "Schindler's List", 
        'Genre':'War', 
        'Director':'Steven Spielberg',
        'Year':'1993',
        'Cast':'Liam Neeson, Ben Kingsley, Ralph Fiennes'
    },
    {
        'id': 'war-ThePianist',
        'name':'The Pianist', 
        'Genre':'War', 
        'Director':'Roman Polanski',
        'Year':'2002',
        'Cast':'Adrien Brody, Frank Finlay'
    }
]