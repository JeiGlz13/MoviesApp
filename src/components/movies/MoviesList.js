import React, { useMemo } from 'react';
import { getMovieByGenre } from '../selectors/getMovieByGenre';
import { MovieCard } from './MovieCard';

export const MoviesList = ({genre}) => {

    const movies =useMemo(() => getMovieByGenre(genre ), [genre])
    

    return (
        <div className = "animate__animated animate__fadeIn w-full flex flex-row flex-wrap justify-around mb-16" >
            {
                movies.map(movie =>(
                    <MovieCard key = {movie.id}
                    {...movie} />
                  
                ))
            } 
        </div>
    )
}
