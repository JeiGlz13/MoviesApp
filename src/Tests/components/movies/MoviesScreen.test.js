import React from 'react';
import {mount} from 'enzyme';
import { MoviesScreen } from '../../../components/movies/MoviesScreen';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Pruebas en <MoviesScreen/>', () => {
    const history = {
        length: 10,
        goBack: jest.fn(),
        push: jest.fn()
    }


   test('Debe mostrar el componente redirect si no hay argumentos en el url', () => {
        const wrapper = mount(<MemoryRouter initialEntries = {['movie']} >
            <MoviesScreen history = {history} />
        </MemoryRouter>);
      expect(wrapper).toMatchSnapshot(); 
      expect(wrapper.find('Redirect').exists()).toBe(true);
   });
    
   test('Debe mostrar un heroe si el parametro existe ', () => {
       const wrapper = mount(
       <MemoryRouter initialEntries = {['/movie/gang-Goodfellas']} >
            <Route path = "/movie/:movieId" component = {MoviesScreen}/>
        </MemoryRouter>);

        expect(wrapper.find('.row').exists()).toBe(true);
   });

   test('Debe rgresar a la pantalla anterior con PUSH', () => {
    const history = {
        length: 1,
        goBack: jest.fn(),
        push: jest.fn()
    }

    const wrapper = mount(
        <MemoryRouter initialEntries = {['/movie/gang-Goodfellas']} >
             <Route path = "/movie/:movieId" component = {(props) => <MoviesScreen history = {history} />}/>
         </MemoryRouter>);

         wrapper.find('button').simulate('click');

         expect(history.push).toHaveBeenCalledWith('/');
         expect(history.goBack).not.toHaveBeenCalled();
   });
   
   test('Debe regresar a la pantalla anterior', () => {
    const wrapper = mount(
        <MemoryRouter initialEntries = {['/movie/gang-Goodfellas']} >
             <Route path = "/movie/:movieId" component = {(props) => <MoviesScreen history = {history} />}/>
         </MemoryRouter>);

         wrapper.find('button').simulate('click');
         expect(history.push).not.toHaveBeenCalled();
         expect(history.goBack).toHaveBeenCalledWith();
   });
   
   test('Debe llamar el redirect si movie no existe', () => {
    const wrapper = mount(
        <MemoryRouter initialEntries = {['/movie/gang-Goodfellas25']} >
             <Route path = "/movie/:movieId" component = {(props) => <MoviesScreen history = {history} />}/>
         </MemoryRouter>);

         expect(wrapper.text()).toBe('');
   });
   
});
