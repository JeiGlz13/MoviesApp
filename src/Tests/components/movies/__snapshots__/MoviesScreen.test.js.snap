// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MoviesScreen/> Debe mostrar el componente redirect si no hay argumentos en el url 1`] = `""`;
